### ISSUE TITLE
Issue with samples

### ISSUE THREAD

**Issue by pmoulon** [Parskatt/DeDoDe#10]
When trying to run the matching sample I have this error
`error: Tensor object has no attribute mT` in `dedode descriptor.py` line 42
- Does it run any bell, did I install something wrong, does the code need any update?

**Comment by Parskatt** [comment]
What pytorch version are you running? mT is transposes the last 2 dimwnsions, not sure when added to pytorch.

**Comment by pmoulon** [comment]
I installed following the setup.py and requirements file. I can double check version later in the week

**Comment by Parskatt** [comment]
Yes the setup doesnt require specific versions. It's tested for python 3.10 with the most recent pytorch (2.0.1). It should work for pytorch down to maybe 1.12 or so.

**Comment by urbste** [comment]
I also had a problem when testing, with python < 3.9. 

Type hinting like this f:tuple["B","H","W"] or tuple[...,2,2] is not supported prior to python (I think 3.9). 
Maybe you could add this requirement somewhere. :-)

**Comment by Parskatt** [comment]
Not sure whats best, I could enforce my own development env, but then some this will make it more difficult to integrate with other frameworks in older versions. Ill think about it.

**Comment by Parskatt** [comment]
I've updated the setup.py with >= 3.9 and torch with >= 1.10 (this is the earliest reference I could find to .mT).
